### ISSUE TITLE
Support parsimonious and six

### ISSUE THREAD

**Issue by mach881040** [brython-dev/brython#1915]
Parsimonious and six are both Pure Python libraries, but both are broken in Brython.

https://github.com/erikrose/parsimonious
https://github.com/benjaminp/six

If you try to run parsimonious in Brython:

```python
from browser import document
import parsimonious

from parsimonious.grammar import Grammar
grammar = Grammar(
"""
bold_text  = bold_open text bold_close
text       = ~"[A-Z 0-9]*"i
bold_open  = "(("
bold_close = "))"
""")
```

We get this stack trace:

```
brython.js:7624 Traceback (most recent call last):
  File http://localhost:8000/#__main__ line 2, in <module>
    import parsimonious
  File http://localhost:8000/parsimonious/__init__.py line 9, in <module>
    from parsimonious.grammar import Grammar, TokenGrammar
  File http://localhost:8000/parsimonious/grammar.py line 13, in <module>
    from parsimonious.expressions import (Literal, Regex, Sequence, OneOf,
  File http://localhost:8000/parsimonious/expressions.py line 16, in <module>
    from parsimonious.nodes import Node, RegexNode
  File http://localhost:8000/parsimonious/nodes.py line 161, in <module>
    class NodeVisitor(with_metaclass(RuleDecoratorMeta, object)):
  File http://localhost:8000/six.py line 884, in __new__
    return meta(name, resolved_bases, d)
  File http://localhost:8000/parsimonious/nodes.py line 142, in __new__
    methods = [v for k, v in iteritems(namespace) if
  File http://localhost:8000/six.py line 616, in iteritems
    return iter(d.items(**kw))
TypeError: items() takes no arguments (1 given)
```

This is because six contains the following constructions, which are valid in Python 3:

```python
    def iterkeys(d, **kw):
        return iter(d.keys(**kw))

    def itervalues(d, **kw):
        return iter(d.values(**kw))

    def iteritems(d, **kw):
        return iter(d.items(**kw))

    def iterlists(d, **kw):
        return iter(d.lists(**kw))

```

We can fix this for Brython:

```python
    def iterkeys(d, **kw):
        return iter(d.keys())

    def itervalues(d, **kw):
        return iter(d.values())

    def iteritems(d, **kw):
        return iter(d.items())

    def iterlists(d, **kw):
        return iter(d.lists())
```

This results in a further error. This error is not related to changing the above code in six. I know, because I first removed six from parsimonious completely, and saw this same error:

```
brython.js:7624 Traceback (most recent call last):
  File http://localhost:8000/#__main__ line 2, in <module>
    import parsimonious
  File http://localhost:8000/parsimonious/__init__.py line 9, in <module>
    from parsimonious.grammar import Grammar, TokenGrammar
  File http://localhost:8000/parsimonious/grammar.py line 478, in <module>
    rule_grammar = BootstrappingGrammar(rule_syntax)
  File http://localhost:8000/parsimonious/grammar.py line 68, in __init__
    exprs, first = self._expressions_from_rules(rules, decorated_custom_rules)
  File http://localhost:8000/parsimonious/grammar.py line 184, in _expressions_from_rules
    spaceless_literal = Regex(r'u?r?"[^"\\]*(?:\\.[^"\\]*)*"',
  File http://localhost:8000/parsimonious/expressions.py line 282, in __init__
    self.re = re.compile(pattern, (ignore_case and re.I) |
TypeError: 'or' not supported between instances of 'Flag' and 'bool'
```

This is due to the following code in parsimonious:

```python
        self.re = re.compile(pattern, (ignore_case and re.I) |
                                      (locale and re.L) |
                                      (multiline and re.M) |
                                      (dot_all and re.S) |
                                      (unicode and re.U) |
                                      (verbose and re.X) |
                                      (ascii and re.A))

```

This code works in Python 3 but not in Brython. We can try changing it:

```python

       flags = None

        for flag in [(ignore_case, re.I),
                     (locale,      re.L),
                     (multiline,   re.M),
                     (dot_all,     re.S),
                     (unicode,     re.U),
                     (verbose,     re.X),
                     (ascii,       re.A)]:

            if flag[0]:

                if flags == None:
                    flags = flag[1]
                else:
                    flags = flags | flag[1]


        if flags is not None:
            print("flags", flags)
            self.re = re.compile(pattern, flags=flags)
        else:
            self.re = re.compile(pattern)

```

However, this results in an error further downstream.

In short, I believe both of these libraries have demonstrated bugs in Brython.

**Comment by PierreQuentel** [comment]
Thanks for the report @mach881040. There were various bug in the re module and in set implementation. The basic example on parsimonious home page now works.
